cdap-spark-core/src/main/java/co/cask/cdap/app/runtime/spark/BasicSparkClientContext.java

```diff
@@ -256,4 +258,13 @@ public PluginProperties getPluginProperties(String pluginId) {
   public SecureStoreData getSecureData(String namespace, String name) throws Exception {
     return sparkRuntimeContext.getSecureData(namespace, name);
   }
+
+  @Override
+  public ProgramState getState() {
+    return state;
+  }
+
+  public void setState(ProgramState state) {
```

Comment:
This method should be package private since it's only called by `SparkRuntimeService`.